atomica/results.py

```diff
@@ -142,6 +142,47 @@ def get_alloc(self, year=None) -> dict:
 
         return self.model.progset.get_alloc(year, self.model.program_instructions)
 
+    def get_equivalent_alloc(self, year=None) -> dict:
+        """
+        Return minimal spending allocation for a given year based on the coverage
+
+        If the result was generated using programs, this method will return the spending
+        on all programs in the requested years.
+
+        :param year: Optionally specify a scalar or list/array of years to return budget values
+                     for. Otherwise, uses all simulation times
+        :return: Dictionary keyed by program name with arrays of spending values
+        """
+
+        if self.model.progset is None:
+            return None
+
+        if year is None:
+            year = self.t
+            
+        prop_coverage = self.get_coverage(quantity='fraction', year=year)
+        num_eligible = self.get_coverage(quantity='eligible', year=year)
+        equivalent_alloc = sc.odict()
+        for prog in prop_coverage.keys():
+            uc = self.model.progset.programs[prog].unit_cost.interpolate(year)
+            pc = sc.dcp(prop_coverage[prog])
+            
+            if self.model.progset.programs[prog].saturation.has_data:
+                sat = self.model.progset.programs[prog].saturation.interpolate(year)
+                
+                #If prop_covered is higher than the saturation then set it to nan (without the error that would happen from np.log)
+                pc[pc>=sat] = np.nan
+            
+                #invert the calculation on the proportional coverage to determine the necessary "costed" coverage
+                pc = -sat * np.log((sat - pc)/(sat + pc))/2.
+            
+            #multiply the proportion of naively costed coverage by the number of actually eligible people (catching the case where number covered would be higher than the number eligible)
+            num_costed_coverage = pc * num_eligible[prog]
+            
+            equivalent_alloc[prog] = uc * num_costed_coverage
```

Comment:
I think we probably need to check the capacity here as well - if `num_costed_coverage` exceeds the capacity constraint (if there is one) then in that case the equivalent alloc also won't yield the desired coverage

Reply:
Yes, you may be correct, and I think this also should just return `nan` if violated?